### ISSUE TITLE
Blazor setDragImage with Drag & Drop

### ISSUE THREAD

**Issue by TDroogers** [dotnet/aspnetcore#45499]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Is there any intention of implementing `e.dataTransfer.setDragImage` in blazor (wasm)? The only issue related to this problem I found does not have a solution or workaround #33158.

`setDragImage` is used in js to change the object you are dragging, see [mozilla.org](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API#define_the_drag_image)

### Describe the solution you'd like

Extend the `DragEventArgs` class with the js function `setDragImage`

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @TDroogers.
I've parked this in the Backlog to accumulate more feedback around this ask. In the meantime, the CustomEventArgs feature should help you unblock: https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-7.0#custom-event-arguments

**Comment by TDroogers** [comment]
I ended up with using JsRuntime. It works, but is probably not ideal for big drag/drop lists and having this issue solved is preferred. I was unable to get custom event arguments working with functions.

xxx.js:

    function addCustomListener(elementId, html)
    {
        document.getElementById(elementId).addEventListener("dragstart", function(e) {
            let crt = this.cloneNode(true);
            crt.style.position = "absolute"; crt.style.top = "auto"; crt.style.left = "-10000px";
            crt.innerHTML = html;
            document.body.appendChild(crt);
            e.dataTransfer.setDragImage(crt, 0, 0);
        }, false);
    }

xxx.razor.cs

    protected override async Task OnAfterRenderAsync(bool firstRender)
        {
            if (firstRender)
            {
                await JsRuntime.InvokeVoidAsync("addCustomListener",$"{Setup.StylingId}", Setup.HtmlContent);
            }
        }

**Comment by jonathanharr** [comment]
Bumping this up, why can't we modify the DragImage using Blazor? It's doable with a full JS interrupt but like @TDroogers said, but it breaks all other functionality. Setting up a custom listener seems to work but it's very limited as to how much you can modify the image itself.
